app/kafka/manager/utils/AdminUtils.scala

```diff
@@ -238,6 +237,21 @@ class AdminUtils(version: KafkaVersion) {
     createOrUpdateTopicPartitionAssignmentPathInZK(curator, topic, newPartitionsReplicaList, update=true, readVersion=readVersion)
   }
 
+  /* Add partitions to multiple topics. After this operation, all topics will have the same number of partitions */
+  def addPartitionsToTopics(curator: CuratorFramework,
+                            topicAndReplicaList: Seq[(String, Map[Int, Seq[Int]])],
+                            newNumPartitions: Int,
+                            brokerList: Seq[Int],
+                            readVersions: Map[String,Int]) {
+    val topicsWithoutReadVersion = topicAndReplicaList.map(x=>x._1).filter{t => !readVersions.contains(t)}
+    checkCondition(topicsWithoutReadVersion.isEmpty, TopicErrors.NoReadVersionFound(topicsWithoutReadVersion.mkString(", ")))
+
+    // topicAndReplicaList is sorted by number of partitions each topic has in order not to start adding partitions if any of requests doesn't work with newNumPartitions
+    for ((topic, replicaList) <- topicAndReplicaList) {
```

Comment:
Since you are already doing the readVersion check, you can do something like this here:

for {
  (topic, replicaList) <- topicAndReplicaList
  readVersion = readVersions(topic)
} {
  ...
}